tests/unit_tests/data_input/layeredmap/test_layered_fence.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+from xtgeo import RegularSurface, Cube, Grid, GridProperty
+from webviz_subsurface.datainput.layeredmap._layered_fence import (
+    LayeredFence,
+)
+
+SURFACE = RegularSurface("tests/data/surface.gri")
+CUBE = Cube("tests/data/seismic.segy")
+FENCESPEC = np.loadtxt("tests/data/polyline.np.gz")
+S_SLICE = np.loadtxt("tests/data/surface_slice.np.gz")
+with open("tests/data/seismic_png.txt", "r") as f:
+    SEISMIC_IMG = f.read()
+GRID = Grid('tests/data/grid.roff')
+GRIDPROP = GridProperty('tests/data/prop.roff')
+with open("tests/data/gridprop_png.txt", "r") as f:
+    GRIDPROP_IMG = f.read()
+
+
+def test_layered_fence_init():
+    fence = LayeredFence(FENCESPEC)
+    assert np.array_equal(fence.fencespec, FENCESPEC)
+    assert fence._surface_layers == []
+    assert fence.bounds == [[0, 0], [0, 0]]
+    assert fence.center == [0, 0]
+
+
+def test_slice_surface():
+    fence = LayeredFence(FENCESPEC)
+    assert np.array_equal(fence.slice_surface(SURFACE.copy()), S_SLICE)
+
+
+def test_set_surface_bounds_and_center():
+    fence = LayeredFence(FENCESPEC)
+    fence.set_bounds_and_center(SURFACE)
+    assert fence.bounds == [
+        [-40.260988980909914, -1653.468683917269],
+        [4489.7725603388635, -1586.0012876253734],
+    ]
+    assert fence.center == [2225.2178792970494, -1618.0241894694848]
```

Comment:
Should we change these types of check to e.g. `np.allclose(fence.center, [2225.2178792970494, -1618.0241894694848])`, just "to be sure"? I'm not enough into C/C++, IEEE 754 and different compilation flags to know if floating point output is deterministic across setups/platform 😕

Reply:
Will do. Already had some problems with these tests failing on different setups.